acvm-repo/acir/src/circuit/black_box_functions.rs

```diff
@@ -9,61 +11,225 @@ use strum_macros::EnumIter;
     EcdsaSecp256k1,
+
     /// Verifies a ECDSA signature over the secp256r1 curve.
+    ///
+    /// Same as EcdsaSecp256k1, but done over another curve.
     EcdsaSecp256r1,
-    /// Performs multi scalar multiplication over the embedded curve.
+
+    /// Multiple scalar multiplication with a variable base/input point (P) of the embedded curve
+    /// - input:
+    ///     points (FieldElement, N) a vector of x and y coordinates of input
+    ///     points `[x1, y1, x2, y2,...]`.
+    ///     scalars (FieldElement, N) a vector of low and high limbs of input
+    ///     scalars `[s1_low, s1_high, s2_low, s2_high, ...]`. (FieldElement, N)
+    ///     For Barretenberg, they must both be less than 128 bits.
+    /// - output: (FieldElement, N) a vector of x and y coordinates of output
+    ///   points `[op1_x, op1_y, op2_x, op2_y, ...]``. Points computed as
+    ///   `s_low*P+s_high*2^{128}*P`
+    ///
+    /// Because the Grumpkin scalar field is bigger than the ACIR field, we
+    /// provide 2 ACIR fields representing the low and high parts of the Grumpkin
+    /// scalar $a$: `a=low+high*2^{128}`, with `low, high < 2^{128}`
     MultiScalarMul,
-    /// Calculates the Keccak256 hash of the inputs.
+
+    /// Computes the Keccak-256 (Ethereum version) of the inputs.
+    /// - inputs: Vector of bytes (FieldElement, 8)
+    /// - outputs: Array of 32 bytes (FieldElement, 8)
     Keccak256,
-    /// Keccak Permutation function of 1600 width
+
+    /// Keccak Permutation function of width 1600
+    /// - TODO: inputs/outputs??
     Keccakf1600,
-    /// Compute a recursive aggregation object when verifying a proof inside another circuit.
-    /// This outputted aggregation object will then be either checked in a top-level verifier or aggregated upon again.
+
+    /// Compute a recursive aggregation object when verifying a proof inside
+    /// another circuit.
+    /// This outputted aggregation object will then be either checked in a
+    /// top-level verifier or aggregated upon again.
+    ///
+    /// **Warning: this opcode is subject to change.**
+    /// Note that the `254` in `(FieldElement, 254)` refers to the upper bound of
+    /// the `FieldElement`.
+    /// - verification_key: Vector of (FieldElement, 254) representing the
+    ///   verification key of the circuit being verified
+    /// - public_inputs: Vector of (FieldElement, 254)  representing the public
+    ///   inputs corresponding to the proof being verified
+    /// - key_hash: one (FieldElement, 254). It should be the hash of the
+    ///   verification key. Barretenberg expects the Pedersen hash of the
+    ///   verification key
+    /// - input_aggregation_object: an optional vector of (FieldElement, 254).
+    ///   It is a blob of data specific to the proving system.
+    /// - output_aggregation_object: Some witnesses returned by the function,
+    ///   representing some data internal to the proving system.
```

Comment:
```suggestion
```
These don't exist anymore

Reply:
We are missing the `proof` input though where we can state then that any backend specific information is expected to be a part of the proof or verification key and handled by the backend.